orc8r/cloud/go/obsidian/swagger/handlers/handlers_test.go

```diff
@@ -73,13 +73,115 @@ func Test_GenerateCombinedSpecHandler(t *testing.T) {
 		Method:         "GET",
 		URL:            testURLRoot,
 		Payload:        nil,
-		Handler:        handlers.GetGenerateCombinedSpecHandler(yamlCommon),
+		Handler:        handlers.GetCombinedSpecHandler(yamlCommon),
 		ExpectedStatus: 200,
 		ExpectedResult: expectedSpec,
 	}
 	tests.RunUnitTest(t, e, tc)
 }
 
+func Test_GenerateSpecHandler(t *testing.T) {
+	e := echo.New()
+	testURLRoot := "/magma/v1"
+
+	commonTag := swagger_lib.TagDefinition{Name: "Tag Common"}
+	commonSpec := swagger_lib.Spec{
+		Tags: []swagger_lib.TagDefinition{commonTag},
+	}
+	yamlCommon := marshalToYAML(t, commonSpec)
+
+	// Fail with invalid service name.
+	tc := tests.Test{
+		Method:         "GET",
+		URL:            testURLRoot,
+		Payload:        nil,
+		ParamNames:     []string{"service"},
+		ParamValues:    []string{"invalid_test_spec_service"},
+		Handler:        handlers.GetSpecHandler(yamlCommon),
+		ExpectedStatus: 404,
+		ExpectedError:  "service not found",
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	// Success with valid service name.
+	tag := swagger_lib.TagDefinition{Name: "Tag 1"}
+	expected := swagger_lib.Spec{
+		Tags: []swagger_lib.TagDefinition{tag, commonTag},
+	}
+
+	// Clean up registry
+	defer registry.RemoveServicesWithLabel(orc8r.SwaggerSpecLabel)
+
+	registerServicer(t, "test_spec_service1", tag)
+
+	tc = tests.Test{
+		Method:         "GET",
+		URL:            testURLRoot,
+		Payload:        nil,
+		ParamNames:     []string{"service"},
+		ParamValues:    []string{"test_spec_service1"},
+		Handler:        handlers.GetSpecHandler(yamlCommon),
+		ExpectedStatus: 200,
+		ExpectedResult: expected,
+	}
+	tests.RunUnitTest(t, e, tc)
+}
+
+func Test_GenerateSpecUIHandler(t *testing.T) {
+	e := echo.New()
+	testURLRoot := "/magma/v1"
+
+	tmpl, err := template.New("test_template.html").Parse("swagger_spec_url: {{.URL}}")
+	assert.NoError(t, err)
+
+	// Fail with invalid service name
+	tc := tests.Test{
+		Method:                 "GET",
+		URL:                    testURLRoot,
+		Payload:                nil,
+		ParamNames:             []string{"service"},
+		ParamValues:            []string{"fake_test_service"},
+		Handler:                handlers.GetUIHandler(tmpl),
+		ExpectedStatus:         404,
+		ExpectedErrorSubstring: "service not found",
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	// Success with empty service name as it should serve the
+	// monolithic Swagger spec
+	tc = tests.Test{
+		Method:         "GET",
+		URL:            testURLRoot,
+		Payload:        nil,
+		ParamNames:     []string{"service"},
+		ParamValues:    []string{""},
+		Handler:        handlers.GetUIHandler(tmpl),
+		ExpectedStatus: 200,
+		ExpectedResult: tests.StringMarshaler("swagger_spec_url: /swagger/v1/spec/"),
+	}
+	tests.RunUnitTest(t, e, tc)
+
+	// Success with valid service name
+	testService2 := "test_spec_service2"
```

Comment:
Nit: rather than creating this variable, just inline the service name literal wherever the variable would have been used